src/main/java/io/cdap/plugin/http/source/common/RawStringPerLine.java

```diff
@@ -61,14 +64,16 @@ public boolean hasNext() {
             isLineRead = true;
             return lastLine != null;
         } catch (IOException e) { // we need to catch this, since hasNext() does not have "throws" in parent
-            throw new RuntimeException("Failed to read line from http page buffer", e);
+            String errorMessage = "Unable to read line from http page buffer";
+            throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+              errorMessage, e.getMessage(), ErrorType.UNKNOWN, true, new IOException(errorMessage));
```

Comment:
why are we creating a new exception when we have instance of IOException here.

Reply:
updated